The overriden print macros take ownership of the arguments

<!--
Please report security concerns to us on the
[AWS vulnerability reporting](http://aws.amazon.com/security/vulnerability-reporting/) page.
-->

<!--
Thank you for filing a bug report! 🐛
Please provide a short summary of the issue, along with the information necessary to replicate.
-->

I tried this code:
<!--
If the reproducing code is small, please post it here.
Otherwise, please attach the relevant files to this issue.
Alternatively, post a link to the repository and branch that exposes the issue.
-->
```rust
#[derive(Debug)]
struct Foo {
    x: i32,
}

#[kani::proof]
fn main() {
    let foo = Foo { x: 5 };
    println!("{:?}", foo);
    let bar = foo;
}
```

using the following command line invocation:

```
kani mv.rs
```

with Kani version: c65b3ec7477

I expected to see this happen: Verification successful

<!--
If Kani crashed, please include a backtrace in the code block by setting
`RUST_BACKTRACE=1` in your environment. E.g. `RUST_BACKTRACE=1 kani ...`
-->
Instead, this happened: 
```
error[E0382]: use of moved value: `foo`
 --> mv.rs:9:15
  |
7 |     let foo = Foo { x: 5 };
  |         --- move occurs because `foo` has type `Foo`, which does not implement the `Copy` trait
8 |     println!("{:?}", foo);
  |                      --- value moved here
9 |     let bar = foo;
  |               ^^^ value used here after move
```

--- a/library/std/src/lib.rs
+++ b/library/std/src/lib.rs
@@ -103,7 +103,7 @@ macro_rules! evaluate_print_args {
         // Evaluate each of the arguments since they may have side effects
         {
             $(
-                $arg;
+                let _ = &$arg;
             )*
         }
     };

--- a/tests/kani/Print/arg_not_moved.rs
+++ b/tests/kani/Print/arg_not_moved.rs
@@ -1,0 +1,20 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+// This test checks that Kani's overridden versions of the print macros do not
+// take ownership of variables passed as arguments
+
+#[derive(Debug)]
+struct Foo {
+    x: i32,
+}
+
+#[kani::proof]
+fn main() {
+    let foo = Foo { x: 5 };
+    // calling `println` with `foo` should not move it
+    println!("{:?}", foo);
+    // make sure reading `foo` does not produce a "use of moved value" error
+    let y = foo.x;
+    assert!(y == 5);
+}

